lib/srv/sessiontracker.go

```diff
@@ -71,24 +72,52 @@ func (s *SessionTracker) Close(ctx context.Context) error {
 	return trace.Wrap(err)
 }
 
-const sessionTrackerExpirationUpdateInterval = apidefaults.SessionTrackerTTL / 6
+const sessionTrackerExpirationUpdateInterval = apidefaults.SessionTrackerTTL / 3
 
 // UpdateExpirationLoop extends the session tracker expiration by 1 hour every 10 minutes
-// until the SessionTracker or ctx is closed.
+// until the SessionTracker or ctx is closed. If there is a failure to write the updated
+// SessionTracker to the backend an exponential backoff retry is attempted until the it
+// has exceeded the original expiration.
 func (s *SessionTracker) UpdateExpirationLoop(ctx context.Context, clock clockwork.Clock) error {
-	ticker := clock.NewTicker(sessionTrackerExpirationUpdateInterval)
-	defer ticker.Stop()
-	return s.updateExpirationLoop(ctx, ticker)
+	return trace.Wrap(s.updateExpirationLoop(ctx, clock))
 }
 
 // updateExpirationLoop is used in tests
-func (s *SessionTracker) updateExpirationLoop(ctx context.Context, ticker clockwork.Ticker) error {
+func (s *SessionTracker) updateExpirationLoop(ctx context.Context, clock clockwork.Clock) error {
+	retry, err := retryutils.NewLinear(retryutils.LinearConfig{
+		Clock:  clock,
+		Max:    3 * time.Minute,
+		Step:   30 * time.Second,
+		Jitter: retryutils.NewHalfJitter(),
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	ticker := clock.NewTicker(sessionTrackerExpirationUpdateInterval)
+	defer func() {
+		if ticker != nil {
+			ticker.Stop()
+		}
+	}()
+
 	for {
 		select {
-		case time := <-ticker.Chan():
-			expiry := time.Add(apidefaults.SessionTrackerTTL)
+		case t := <-ticker.Chan():
+			expiry := t.Add(apidefaults.SessionTrackerTTL)
 			if err := s.UpdateExpiration(ctx, expiry); err != nil {
-				return trace.Wrap(err)
+				// stop and reset the ticker so that it doesn't
+				// keep accumulating ticks while we are retrying
+				ticker.Stop()
+				ticker = nil
```

Comment:
Can you reset the ticker you already have instead of nilling it out and creating a new one?